typing/parmatch.mli

```diff
@@ -35,8 +35,13 @@ val const_compare : constant -> constant -> int
 
 val le_pat : pattern -> pattern -> bool
 val le_pats : pattern list -> pattern list -> bool
-val compat : pattern -> pattern -> bool
-val compats : pattern list -> pattern list -> bool
+
+(* Exported compatibility,
+   "may_compat p q" returns true when p and q never admit a common instance;
+   returns true when they may have a common instance. *)
+val may_compat : pattern -> pattern -> bool
+val may_compats : pattern list -> pattern list -> bool
```

Comment:
Considering that `may_compat` and `may_compats` are not used (and indeed shouldn't be used) in parmatch, instead of defining them in parmatch and exporting them, could we export the functor and define the functions in matching.ml?

Reply:
I'd rather leave this to a later refactoring, such as the one you have started. The functor belongs to some utility module. I remember that have created such a utility module (for printing).

Reply:
The functor doesn't seem to belong to an external utility module any more than `may_compat` itself. And doing the change now instead of later reduces the probability of people using the wrong function by mistake.

Your choice. :)